packages/blade/src/components/Link/_decisions/decisions.md

```diff
@@ -0,0 +1,64 @@
+
+
+## Internal BaseLink Component
+Internal component that exposes certain extra props to enable creation of Component Level Action Link like `positive`, `negative`, `neutral`, `information`, `notice`
+
+### BaseLink API
+| Prop | Type | Required | Default | Description |
+|---|---|---|---|---|
+| variant | `anchor`, `button` | *No* | `anchor` | The variant of the Link to be rendered.<br><br>Note: `button` will look like an inline Link component (no spacing; can be rendered alongside other text) but will be rendered as a `button` element. |
+| intent |  `positive`,`negative`, `notice`, `information`, `neutral` | *No* | undefined | The intent of the link to be rendered.<br><br>  Note: <br> - We will use this to set the intent for internal Blade components that have positive, negative, notice, info, or neutral link buttons like an Alert component. |
+| contrast | `low`, `high` | *No* | `low` | The contrast of the link to be rendered.<br><br>  Note: We need this for internal Blade components since a `positive` intent could have a `high` as well as a `low` contrast link. |
+| children | `string` | When `icon` is passed: *No*<br>When `icon` is not passed: *Yes*   | undefined | The text to be rendered within the link. |
+| icon | `Icon` | When `children` is passed: *No*<br>When `children` is not passed: *Yes* | undefined | The Blade `Icon` component to be rendered within the link. |
+| iconPosition | `left`, `right` | *No* | `right` | The position of the rendered icon. |
+| isDisabled | `boolean` | *No* | `false` | Control whether the link is disabled or not.<br><br>Note: This prop is only valid for `button` variant. |
+| onClick | `Function` | No | undefined | The function to be called when the link is clicked. |
+| href | `string` | No | undefined | The href for the link component.<br><br>Note: This prop is only valid for `anchor` variant. |
+| target | `string` | No | undefined | The target for the link component.<br><br>Note: This prop is only valid for `anchor` variant. |
+| rel | `string` | No | undefined | The rel for the link component.<br><br>Note: This prop is only valid for `anchor` variant. |
+| accessibilityLabel | `string` | No | undefined | The `aria-label` (web) & `accessibilityLabel` (native) of the link component. |
+
+
+## Link Component
+A Link component that can act as a `<button>` as well as an `<anchor>` element. This will be the component exported to the consumers. This is a wrapper on top of `BaseLink` component to avoid exposing `intent` & `contrast` to consumers.
+
+### Link API
+
+| Prop | Type | Required | Default | Description |
+|---|---|---|---|---|
+| variant | `anchor`, `button` | *No* | `anchor` | The variant of the Link to be rendered.<br><br>Note: `button` will look like an inline Link component (no spacing; can be rendered alongside other text) but will be rendered as a `button` element. |
```

Comment:
Better if we make it explicit otherwise it could lead to unintended changes for the consumer.

Reply:
I would recommend keeping the API surface as lean as possible.

Reply:
Yes, but this is a very critical differentiation on whether it will be rendered as an `anchor` element or a `button` element. 

For example, a user passes `onClick` and decides to handle link navigation by themselves by adding `e.preventDefault()` on the onClick and then setting `window.location.href`. In this case the consumer is expecting an `a` tag to be rendered but hasn't passed `href` cuz they have decided to handle it themselves. But since we internally decided to render it as a `button` since `href` is missing, the consumer will end up having an unintended `button` element instead of a `a` element. 

This is why I think we should let the consumer explicitly handle the switch between `a` & `button`

Reply:
if the consumer wants to do something nasty they'll figure out anyway to do it. this seems to be a very rare edge case, as an API designer i would not tradeoff a new prop and learning curve just for such a use case. even if we have a variant(which is not a required prop) consumers can still achieve the use case you have mentioned because they didn't care enough to go through the documentation.

Reply:
I see the point chaitanya is making. But if consumer wants to do that "nasty" navigation they can pass it like this

<Link href="#" onClick={e => e.preventDefault()} /> 

> Also there's nothing called a link without href. If you don't pass href to link. It renders like a plain text :lol:

Reply:
we can rather add a TS check which tells that at least one of `href` or `onClick` is required and then we'll be all good